internal/os/volume/api.go

```diff
@@ -109,6 +110,27 @@ func (VolAPIImplementor) ResizeVolume(volumeID string, size int64) error {
 		finalSize = size
 	}
 
+	cmd = fmt.Sprintf("Get-Volume -UniqueId \"%s\" | Get-partition | Select Size", volumeID)
+	out, err = runExec(cmd)
+
+	if err != nil || len(out) == 0 {
+		return fmt.Errorf("error getting size of the partition from moun. cmd %s, output: %s, error: %v", cmd, string(out), err)
+	}
+
+	var getPartitionSizing map[string]int64
+	outString = string(out)
+	err = json.Unmarshal([]byte(outString), &getPartitionSizing)
+	if err != nil {
+		return fmt.Errorf("out %v outstring %v err %v", out, outString, err)
+	}
+
+	currentSize := getPartitionSizing["Size"]
+
+	//if the partition's size is already the size we want this is a noop, just return
+	if currentSize >= finalSize {
+		return nil
```

Comment:
add a logging here?

Reply:
so far no logging has been done on any of the api in internal/os, I am unsure if that would break a convention